parsons/databases/bigquery/bigquery.py

```diff
@@ -0,0 +1,968 @@
+                "mode": "REQUIRED"
+            },
+            {
+                "name": column_name,
+                "field_type": [INTEGER, STRING, FLOAT, etc.],
+                "default_value_expression": CURRENT_TIMESTAMP()
+            }
+        ]
+        ```
+
+    `Returns`:
+        List of instantiated `SchemaField` objects
+    """
+
+    return [bigquery.SchemaField(**x) for x in schema_definition]
+
+
+class BigQuery(DatabaseConnector):
+    """
+    Class for querying BigQuery table and returning the data as Parsons tables.
+
+    This class requires application credentials in the form of a json. It can be passed
+    in the following ways:
+
+    * Set an environmental variable named ``GOOGLE_APPLICATION_CREDENTIALS`` with the
+      local path to the credentials json.
+
+      Example: ``GOOGLE_APPLICATION_CREDENTALS='path/to/creds.json'``
+
+    * Pass in the path to the credentials using the ``app_creds`` argument.
+
+    * Pass in a json string using the ``app_creds`` argument.
+
+    Args:
+        app_creds: str
+            A credentials json string or a path to a json file. Not required
+            if ``GOOGLE_APPLICATION_CREDENTIALS`` env variable set.
+        project: str
+            The project which the client is acting on behalf of. If not passed
+            then will use the default inferred environment.
+        location: str
+            Default geographic location for tables
+    """
+
+    def __init__(self, app_creds=None, project=None, location=None):
+        self.app_creds = app_creds
+
+        setup_google_application_credentials(app_creds)
+
+        self.project = project
+        self.location = location
+
+        # We will not create the client until we need to use it, since creating the client
+        # without valid GOOGLE_APPLICATION_CREDENTIALS raises an exception.
+        # This attribute will be used to hold the client once we have created it.
+        self._client = None
+
+        self._dbapi = dbapi
+
+        self.dialect = "bigquery"
+
+    @property
+    def client(self):
+        """
+        Get the Google BigQuery client to use for making queries.
+
+        `Returns:`
+            `google.cloud.bigquery.client.Client`
+        """
+        if not self._client:
+            # Create a BigQuery client to use to make the query
+            self._client = bigquery.Client(project=self.project, location=self.location)
+
+        return self._client
+
+    @contextmanager
+    def connection(self):  # TODO: Is this worth doing given the transaction thing?
+        """
+        Generate a BigQuery connection.
+        The connection is set up as a python "context manager", so it will be closed
+        automatically when the connection goes out of scope. Note that the BigQuery
+        API uses jobs to run database operations and, as such, simply has a no-op for
+        a "commit" function. If you would like to manage transactions, please use
```

Comment:
Is this connection even worth having on the BigQuery connector because of the state issues?

Reply:
Bumping this because I'm not sure if it was ever resolved.

Reply:
I think given the error checks we've put in, it's probably not worth deprecating since this was already on the class before